### DISCUSSION TITLE
How to use $apollo command in vuex store actions?

### DISCUSSION THREAD

**Discussion by pschaub** [Akryum/vue-apollo#161] [General]
How can I use the $apollo query/mutation command in vuex store actions?

I have very general user data (e.g. name, permissions, ..) which should be fetched one time after login and stored in vuex. For this I want to use vuex actions to perform the apollo query command.

How can I get access to the $apollo.query() command in vuex actions?

**Comment by Austio** [comment]
https://github.com/Akryum/vue-apollo/issues/7 please read through this, should give you a good idea of what/how to do this. 

With apollo2 they dropped the hard dependency on redux, so this will be much easier to implement once vue-apollo 3 is released.

**Comment by joe-re** [comment]
I'm using $apollo that is stacked on `store.dispatch` payload like following.

- vue component
```
export default Vue.extend({
  //...
  mounted() {
    this.$store.dispatch({ type: 'app/initialize', apolloClient: this.$apollo })
  }
})
```

- actions
```
  actions: {
    async initialize(context, payload) {
        const result = await payload.apolloClient.mutate({
          mutation: AUTHENTICATE_USER,
          variables: { accessToken: localStorage.getItem('auth0_access_token') },
        })
       //...
    },
```

It works for me, but ugly a bit...

@Akryum 
You said it would be resolved by vuex plugin in bellow comment.
https://github.com/Akryum/vue-apollo/issues/7#issuecomment-251580141
Are you planning to provide plugin of vuex in our project?

**Comment by samuelpatro** [comment]
Better way is to export apollo-client and vue-apollo in 2 separate files, then you can simply import apollo-client in your store like `import apollo from '../apollo/myapollo.js'` then you can use it without any dispatching

**Comment by 1clickpersianas** [comment]
@Samuell1 Any real world example?

**Comment by c0h1b4** [comment]
No
